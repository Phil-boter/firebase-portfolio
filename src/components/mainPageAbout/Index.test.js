import { shallow } from "enzyme";

import { Router, BrowserRouter } from "react-router-dom";

import { createMemoryHistory } from "history";

import { render, fireEvent } from "@testing-library/react";

import MainPageAbout from "./Index.js";

let language = "en";

describe("MainPageAbout Component", () => {
    it("should render correctly and match snapshot", () => {
        const { container } = render(
            <BrowserRouter>
                <MainPageAbout language={language} />
            </BrowserRouter>
        );
        expect(container).toMatchSnapshot();
    });
    it("should render MainPageAbout correctly", () => {
        const wrapper = shallow(<MainPageAbout language={language} />);
        expect(wrapper.find(".section-main-about").length).toBe(1);
    });
    it("should have a header GET TO KNOW ME uppercase", () => {
        const wrapper = shallow(<MainPageAbout language={language} />);
        expect(wrapper.find("h1").text()).toEqual("GET TO KNOW ME!");
    });
    it("should render about page main image", () => {
        const wrapper = shallow(<MainPageAbout language={language} />);
        wrapper.find(".section-inner-image");
        expect(wrapper.find(".section-inner-image").length).toBe(1);
        expect(wrapper.find(".section-inner-image").prop("src")).toEqual(
            "/assets/content-pixie-iXRVqQtBa_8-unsplash.jpg"
        );
    });
    it("should render button, that redirect to about if Clicked", async () => {
        const history = createMemoryHistory();
        history.push = jest.fn();

        const { container } = render(
            <Router history={history}>
                <MainPageAbout language={language} />
            </Router>
        );
        fireEvent.click(container.querySelector(".main-page-about-link"));

        expect(history.push).toHaveBeenCalledWith("/about");
    });
});
