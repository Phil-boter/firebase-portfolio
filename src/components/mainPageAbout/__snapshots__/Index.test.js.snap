// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPageAbout Component should render correctly and match snapshot 1`] = `
<div>
  <section
    class="section-main-about"
  >
    <div
      class="main-about-article-wrapper"
    >
      <div
        class="main-header"
      >
        <h1>
          GET TO KNOW ME!
        </h1>
      </div>
      <article
        class="about-article"
      >
        <p>
          Do you want to know more about me or professional background? Then have a look at my about page.
        </p>
        <div
          class="about-button-container"
        >
          <a
            class="main-page-about-link"
            href="/about"
          >
            <button
              class="button about-button"
            >
              Visit About
              <img
                alt="arrow points right"
                data-testid="main-about-image"
                src="/assets/arrow1.png"
              />
            </button>
          </a>
        </div>
      </article>
    </div>
    <div
      class="section-image"
    >
      <img
        alt=""
        class="section-inner-image"
        src="/assets/content-pixie-iXRVqQtBa_8-unsplash.jpg"
      />
    </div>
  </section>
</div>
`;
